### PR TITLE
Add Unciv app

### PR DESCRIPTION
**Pull request by touhidurrr**
Add Unciv 3.18.10-patch1 app

### CODE REVIEW

File: apps/Unciv/description [hunk 1]
```diff
@@ -0,0 +1,3 @@
+An open-source, mod-friendly Android+Desktop remake of Civ V
+
+To run: Menu -> Games -> Sublime Text
```
Comment (Botspot) [inline apps/Unciv/description:3]: Could you fix the "To run" entry?

File: apps/Unciv/install-32 [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+### make directory ~/Unciv if not available
+if [[ ! -d ~/Unciv ]]; then
+  mkdir ~/Unciv || error 'Failed to make folder $HOME/Unciv!'
+fi
+
+### Add LICENSE from Github if doesn't exist
+if [[ ! -f ~/Unciv/LICENSE ]]; then
+  wget -O ~/Unciv/LICENSE ~ https://raw.githubusercontent.com/yairm210/Unciv/master/LICENSE || error 'Failed to make folder $HOME/Unciv!'
```
Comment (Botspot) [inline apps/Unciv/install-32:10]: The error message on this line does not make sense. Also, why is there a `~` character in this wget command?

File: apps/Unciv/install-32 [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  # Extract jre folder
+  tar --overwrite -xvf ~/zulu8.58.0.13-ca-jdk8.0.312-linux_aarch32hf.tar.gz --wildcards */jre || error 'Failed to ectract jre for Unciv!'
+
+  # move jre to ~/Unciv
+  mv ~/zulu8.58.0.13-ca-jdk8.0.312-linux_aarch32hf/jre ~/Unciv/jre || error "Failed to move jre for Unciv to $HOME/Unciv!"
+fi
+
+## remove artifacts
+if [[ -f ~/zulu8.58.0.13-ca-jdk8.0.312-linux_aarch32hf.tar.gz ]]; then
+  rm ~/zulu8.58.0.13-ca-jdk8.0.312-linux_aarch32hf.tar.gz || echo 'Failed to remove jre artifatcs! Skipping ...'
```
Comment (Botspot) [inline apps/Unciv/install-32:35]: You could simplify this by removing the `if` statement and using `rm -f`.

File: apps/Unciv/install-32 (hunk 2 above)
Comment (touhidurrr) [reply apps/Unciv/install-32]: Unintended. Fixed in https://github.com/touhidurrr/pi-apps/commit/ff9b43b14eb696080bd03d096f4f9c2d43f42236.

File: apps/Unciv/install-32 (hunk 3 above)
Comment (touhidurrr) [reply apps/Unciv/install-32]: Ok, yes, it works. Both for folder and file. Fixed in https://github.com/touhidurrr/pi-apps/commit/00638217a58f8323b8cd04dea85caed6463a73ed.

File: apps/Unciv/uninstall [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+rm -f ~/.local/share/applications/unciv.desktop
+rm -rf ~/Unciv
+purge_packages || exit 1
```
Comment (touhidurrr) [inline apps/Unciv/uninstall:5]: @Botspot fixed the mistakes that you pointed out. However, I am wondering what this line does. I copied it from another app entry that I was using as a reference. But don't know what it does.

**Comment by touhidurrr** [comment]
@Botspot fixed the mistakes that you pointed out. However, I am wondering what this line does. I copied it from another app entry that I was using as a reference. But don't know what it does.
https://github.com/touhidurrr/pi-apps/blob/00638217a58f8323b8cd04dea85caed6463a73ed/apps/Unciv/uninstall#L5

File: apps/Unciv/description (hunk 1 above)
Comment (touhidurrr) [reply apps/Unciv/description]: Fixed in https://github.com/Botspot/pi-apps/pull/1274/commits/277c85b4483299b52082f17881fa6c5667e03748.

**Comment by touhidurrr** [comment]
Also since the Unciv app is updated very frequently, I want to add some kind of auto update mechanism to it. Can I do it?

**Comment by cycool29** [comment]
> However, I am wondering what this line does. I copied it from another app entry that I was using as a reference. But don't know what it does.
https://github.com/touhidurrr/pi-apps/blob/00638217a58f8323b8cd04dea85caed6463a73ed/apps/Unciv/uninstall#L5


That is to remove APT packages installed when installing the app.
Since the app don't install packages when install, so that line is unneeded.

**Comment by touhidurrr** [comment]
> That is to remove APT packages installed when installing the app.
Since the app don't install packages when install, so that line is unneeded.

Very vague. How does it figure out if any apt package was installed when installing the app?

**Comment by cycool29** [comment]
> Very vague. How does it figure out if any apt package was installed when installing the app? 

https://github.com/Botspot/pi-apps/blob/master/DOCUMENTATION.md#purpose-1

> Should I remove it or not?

Yes.

**Pull request closed without merging**